src/dynarec/arm64/dynarec_arm64_66f30f.c

```diff
@@ -63,19 +115,21 @@ uintptr_t dynarec64_66F30F(dynarec_arm_t* dyn, uintptr_t addr, uintptr_t ip, int
             nextop = F8;
             GETEW(x1, 0);
             GETGW(x2);
+            TSTxw_REG(ed, ed);
             IFX(X_CF) {
-                TSTxw_REG(ed, ed);
                 CSETw(x3, cEQ);
                 BFIw(xFlags, x3, F_CF, 1);  // CF = is source 0?
             }
             LSLw_IMM(ed, ed, 16);
             CLZw(gd, ed);
+            MOV32w(x3, 16);
```

Comment:
another solution would be the set bit 15 as 1, with an OR_mask, but his solution works too and fixes the issue

Reply:
Or just not LSL and substract 16 to the result. might be even simpler.